homeassistant/components/hassmpris/media_player.py

```diff
@@ -0,0 +1,625 @@
+
+    component_data = cast(
+        HassmprisData,
+        hass.data[DOMAIN][config_entry.entry_id],
+    )
+    mpris_client = component_data.client
+    manager = MPRISCoordinator(
+        hass,
+        config_entry,
+        mpris_client,
+        async_add_entities,
+    )
+    await manager.start()
+    component_data.entity_manager = manager
+
+    async def _async_stop_manager(*unused_args: Any) -> None:
+        # The following is a very simple trick to delete the
+        # reference to the manager once the manager is stopped
+        # once via this mechanism.
+        # That way, if the manager is stopped because the entry
+        # was unloaded (e.g. integration deleted), this will
+        # not try to stop the manager again.
+        if component_data.entity_manager:
+            _LOGGER.debug("Stopping entity manager")
+            await component_data.entity_manager.stop()
+            _LOGGER.debug("Entity manager stopped")
+            component_data.entity_manager = None
+
+    component_data.unloaders.append(_async_stop_manager)
+
+
+class HASSMPRISEntity(CoordinatorEntity["MPRISCoordinator"], MediaPlayerEntity):
+    """Represents an MPRIS media player entity."""
+
+    _attr_device_class = MediaPlayerDeviceClass.TV
+    _attr_supported_features = SUPPORTED_MINIMAL
+    _attr_playback_rate: float = 1.0
+    _attr_should_poll: bool = False
+    _attr_has_entity_name = True
+
+    coordinator_context: str
+
+    def __init__(
+        self,
+        coordinator: MPRISCoordinator,
+        context: str,
+        integration_id: str,
+    ) -> None:
+        """Initialize the entity.
+
+        Arguments:
+          coordinator: the coordinator handling this entity
+          context: the name / unique identifier of the player,
+                   also used as the coordinator context
+          integration_id: unique identifier of the integration
+        """
+        super().__init__(coordinator, context=context)
+        self._integration_id = integration_id
+        self._metadata: dict[str, Any] = {}
+
+    def _debug(self, format_string: str, *args: Any) -> None:
+        _LOGGER.debug("%s: " + format_string, self.name, *args)
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available.  Overridden from Coordinator."""
+        return self._attr_available
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        if self.coordinator_context not in self.coordinator.data:
+            # This player has been removed.  Ignore this.
+            return
+
+        updated = False
+
+        def reset_player_attributes() -> None:
+            self._attr_media_position = None
+            self._attr_media_position_updated_at = dt_util.utcnow()
+            self._attr_media_duration = None
+            self._attr_playback_rate = 1.0
+            self._metadata = {}
+
+        while self.coordinator.data[self.coordinator_context]:
```

Comment:
Yes.  Not just because that's how the implementation may work (e.g. you get a general player update and a Seeked player update, these are different messages) but also because we coalesce rapid changes within 20ms.